### PR TITLE
EPMRPP-83121 || Implement the API KEYS tab view (Create/Revoke modals)

### PR DESCRIPTION
**Pull request by Vadim73i**

### CODE REVIEW

File: app/src/pages/inside/profilePage/modals/revokeApiKeyModal/revokeApiKeyModal.jsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  const [loading, setLoading] = useState(false);
+
+  const onRevoke = (closeModal) => {
+    setLoading(true);
+    dispatch(
+      deleteApiKeysAction({
+        apiKeyId: id,
+        successMessage: formatMessage(messages.successNotification),
+        errorMessage: formatMessage(messages.notificationFail),
+        onSuccess: () => closeModal(),
```
Comment (AmsterGet) [inline app/src/pages/inside/profilePage/modals/revokeApiKeyModal/revokeApiKeyModal.jsx:69]:
```suggestion
        onSuccess: closeModal,
```

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

File: app/src/controllers/user/sagas.js [hunk 2]
```diff
@@ -154,35 +158,77 @@ function* generateApiToken({ payload = {} }) {
         }),
       );
     }
-  } catch (err) {
+    yield put(fetchApiKeysAction());
```
Comment (Bam6ycha) [inline app/src/controllers/user/sagas.js:161]: I am not sure, that we need to fetch all APIKeys. As I understand correctly we already have has all keys.
We store all keys in our store, and when we create new key, we can just add new key to the our store.

File: app/src/controllers/user/sagas.js [hunk 3]
```diff
@@ -154,35 +158,77 @@ function* generateApiToken({ payload = {} }) {
...
+    onSuccess();
+    if (successMessage) {
+      yield put(
+        showNotification({
+          message: successMessage,
+          type: NOTIFICATION_TYPES.SUCCESS,
+        }),
+      );
+    }
+    yield put(fetchApiKeysAction());
```
Comment (Bam6ycha) [inline app/src/controllers/user/sagas.js:208]: Here is the same

File: app/src/controllers/user/sagas.js [hunk 4]
```diff
@@ -158,7 +160,10 @@ function* addApiKey({ payload = {} }) {
         }),
       );
     }
-    yield put(fetchApiKeysAction());
+    const apiKeys = yield select((state) => apiKeysSelector(state));
+    const updatedApiKeys = [...apiKeys];
+    updatedApiKeys.unshift({ id, name: newName, created_at });
+    yield put(setApiKeysAction(updatedApiKeys));
```
Comment (AmsterGet) [inline app/src/controllers/user/sagas.js:166]: Please move this logic to the reducer.
Create a separate action handlers to address such actions.

File: app/src/controllers/user/sagas.js [hunk 5]
```diff
@@ -149,7 +148,10 @@ function* addApiKey({ payload = {} }) {
         name,
       },
     });
-    onSuccess(response.api_key);
+
+    // eslint-disable-next-line camelcase
+    const { id, name: newName, created_at, api_key } = response;
```
Comment (AmsterGet) [inline app/src/controllers/user/sagas.js:153]: You can use the name from the `payload` as they are the same.

File: app/src/controllers/user/sagas.js [hunk 6]
```diff
@@ -205,7 +210,9 @@ function* deleteApiKey({ payload = {} }) {
         }),
       );
     }
-    yield put(fetchApiKeysAction());
+    const apiKeys = yield select((state) => apiKeysSelector(state));
+    const updatedApiKeys = apiKeys.filter((key) => key.id !== apiKeyId);
+    yield put(setApiKeysAction(updatedApiKeys));
```
Comment (AmsterGet) [inline app/src/controllers/user/sagas.js:215]: Move to the reducer.

**Review by AmsterGet — CHANGES_REQUESTED** [review]

File: app/src/pages/inside/profilePage/apiKeys/apiKeysBlock/apiKeysBlock.jsx [hunk 7]
```diff
@@ -77,13 +77,16 @@ export const ApiKeysBlock = ({ apiKeys }) => {
...
                 <div className={cx('revoke-col')}>
-                  <button
-                    className={cx('revoke-button')}
+                  <GhostButton
                     onClick={() => onRevokeClick(key)}
                     title={formatMessage(messages.revoke)}
+                    transparentBorder
+                    transparentBackground
+                    transparentBorderHover
+                    color={'red'}
```
Comment (Bam6ycha) [inline app/src/pages/inside/profilePage/apiKeys/apiKeysBlock/apiKeysBlock.jsx:86]: we can omit curly braces here.

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

File: app/src/controllers/user/sagas.js [hunk 8]
```diff
@@ -180,8 +179,8 @@ function* addApiKey({ payload = {} }) {
 function* fetchApiKeys() {
   const user = yield select(userInfoSelector);
   try {
-    const response = yield call(fetch, URLS.apiKeys(user.userId));
-    yield put(setApiKeysAction(response));
+    const response = yield call(fetch, URLS.apiKeys(user.id));
+    yield put(setApiKeysAction(response.items.sort((a, b) => b.created_at - a.created_at)));
```
Comment (Bam6ycha) [inline app/src/controllers/user/sagas.js:183]: can we move this logic to reducer?

**Review by Bam6ycha — CHANGES_REQUESTED** [review]

**Review by AmsterGet — APPROVED** [review]

**Review by AmsterGet — APPROVED** [review]

**Pull request merged**